src/main/java/frc/robot/poseestimator/helpers/RobotHeadingEstimator.java

```diff
@@ -0,0 +1,48 @@
+package frc.robot.poseestimator.helpers;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.interpolation.TimeInterpolatableBuffer;
+import frc.robot.poseestimator.PoseEstimatorMath;
+
+import java.util.Optional;
+
+public class RobotHeadingEstimator {
+
+	private final TimeInterpolatableBuffer<Rotation2d> gyroAngleInterpolator;
```

Comment:
perhaps you should clarify in the name of the the interpolator that it's unoffseted gyro angle and can't be rawly used for anything outside of calculating offsets.